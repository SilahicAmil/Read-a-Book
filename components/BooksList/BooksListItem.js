import {
  Image,
  Pressable,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from "react-native";

import { ListItem } from "@rneui/themed";

const BooksListItem = ({
  title,
  authorFirst,
  authorLast,
  onPress,
  description,
  ratings,
}) => {
  return (
    <>
      <ScrollView style={styles.container}>
        <View style={styles.shadowContainer}>
          <Pressable
            onPress={onPress}
            style={({ pressed }) => (pressed ? styles.pressed : null)}
          >
            <ListItem containerStyle={styles.innerContainer}>
              <Image
                alt={title + "cover"}
                source={{
                  uri: "https://images.unsplash.com/photo-1528459105426-b9548367069b?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=664&q=80",
                }}
                style={styles.image}
              />
              <View>
                <ListItem.Content style={styles.contentContainer}>
                  <ListItem.Title>{title}</ListItem.Title>
                  <ListItem.Subtitle>
                    by {authorFirst}&nbsp;{authorLast}
                  </ListItem.Subtitle>
                </ListItem.Content>
                <Text style={styles.descriptionText}>
                  {description.slice(0, 125)}...
                </Text>
              </View>
            </ListItem>
          </Pressable>
        </View>
      </ScrollView>
    </>
  );
};

export default BooksListItem;

const styles = StyleSheet.create({
  container: {
    backgroundColor: "white",
    marginTop: 12,
    marginBottom: 12,
    borderRadius: 8,
    height: 100,
    display: "flex",
    width: "95%",
    marginLeft: "auto",
    marginRight: "auto",
  },
  shadowContainer: {
    backgroundColor: "white",
    shadowColor: "black",
    shadowOffset: { width: 1, height: 8 },
    shadowOpacity: 0.4,
    elevation: 12,
  },
  innerContainer: {
    backgroundColor: "#FBF8F2",
    width: "100%",
    height: "100%",
    borderRadius: 8,
  },
  image: {
    width: "20%",
    height: "100%",
    borderRadius: 8,
  },
  contentContainer: {
    width: "100%",
    height: "100%",
    display: "flex",
    flexDirection: "column",
  },
  descriptionText: {
    width: "38%",

    fontSize: 12,
    fontFamily: "Kantumury-Pro",
  },
  pressed: { opacity: 0.5 },
});
