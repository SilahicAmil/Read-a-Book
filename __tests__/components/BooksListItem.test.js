import BooksListItem from "../../components/BooksList/BooksListItem";
import renderer from "react-test-renderer";

describe("BooksListItem Component", () => {
  it("should render correctly", () => {
    description = " Lorem ipsum dolor, sit amet consectetur adipisicing elit. ";
    const tree = renderer
      .create(<BooksListItem description={description} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
