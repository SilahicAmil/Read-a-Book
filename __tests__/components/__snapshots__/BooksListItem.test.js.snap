// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksListItem Component should render correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 8,
      "display": "flex",
      "height": 100,
      "marginBottom": 12,
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 12,
      "width": "95%",
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "white",
          "elevation": 12,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 8,
            "width": 1,
          },
          "shadowOpacity": 0.4,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FBF8F2",
                "borderColor": "#bcbbc1",
                "borderRadius": 8,
                "flexDirection": "row",
                "height": "100%",
                "padding": 14,
                "width": "100%",
              }
            }
            testID="RNE__LISTITEM__padView"
          >
            <Image
              alt="undefinedcover"
              source={
                {
                  "uri": "https://images.unsplash.com/photo-1528459105426-b9548367069b?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=664&q=80",
                }
              }
              style={
                {
                  "borderRadius": 8,
                  "height": "100%",
                  "width": "20%",
                }
              }
            />
            <View
              style={
                {
                  "paddingLeft": 16,
                }
              }
            />
            <View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "flex": 1,
                      "justifyContent": "center",
                    },
                    {
                      "display": "flex",
                      "flexDirection": "column",
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                theme={
                  {
                    "colors": {
                      "background": "#ffffff",
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": {
                        "android": {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "searchBg": "#dcdce1",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ad1457",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                    "spacing": {
                      "lg": 12,
                      "md": 8,
                      "sm": 4,
                      "xl": 24,
                      "xs": 2,
                    },
                  }
                }
              >
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "backgroundColor": "transparent",
                      "color": "#242424",
                      "fontSize": 17,
                    }
                  }
                  testID="listItemTitle"
                />
                <Text
                  accessibilityRole="text"
                  style={
                    {
                      "backgroundColor": "transparent",
                      "color": "#242424",
                      "fontSize": 15,
                    }
                  }
                  testID="listItemTitle"
                >
                  by 
                  Â 
                </Text>
              </View>
              <Text
                style={
                  {
                    "fontFamily": "Kantumury-Pro",
                    "fontSize": 12,
                    "width": "38%",
                  }
                }
              >
                 Lorem ipsum dolor, sit amet consectetur adipisicing elit. 
                ...
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
